unixfs/examples/get.rs

```diff
@@ -70,49 +70,44 @@ fn walk(blocks: ShardedBlockStore, start: &Cid) -> Result<(), Error> {
-                    if segment.is_first() {
-                        // this is set on the root block, no actual bytes are present for multiblock
-                        // files
-                    }
-
-                    if segment.is_last() {
-                        let mode = md.mode().unwrap_or(0o0644) & 0o7777;
-                        let (seconds, _) = md.mtime().unwrap_or((0, 0));
-                        println!("f {:o} {:>12} {:>16} {:?}", mode, seconds, size, path);
-                    }
+        match walker.inspect(&buf, &mut cache)? {
+            ContinuedWalk::File(
+                segment,
+                Entry::Metadata(MetadataEntry::File(.., path, md, size)),
+            ) => {
+                if segment.is_first() {
+                    // this is set on the root block, no actual bytes are present for multiblock
+                    // files
                 }
-                item.into_inner()
-            }
-            ContinuedWalk::Directory(item) => {
-                // presense of metadata can be used to determine if this is the first apperiance of
-                // a directory by looking at the metadata: sibling hamt shard buckets do not have
-                // metadata.
-                if let Entry::Metadata(metadata_entry) = item.as_entry() {
-                    let metadata = metadata_entry.metadata();
-                    let path = item.as_entry().path();
-                    let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
-                    let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
-                    println!("d {:o} {:>12} {:>16} {:?}", mode, seconds, "-", path);
+
+                if segment.is_last() {
+                    let mode = md.mode().unwrap_or(0o0644) & 0o7777;
+                    let (seconds, _) = md.mtime().unwrap_or((0, 0));
+                    println!("f {:o} {:>12} {:>16} {:?}", mode, seconds, size, path);
                 }
-                item.into_inner()
             }
-            ContinuedWalk::Symlink(bytes, item) => {
-                if let Entry::Metadata(metadata_entry) = item.as_entry() {
-                    let metadata = metadata_entry.metadata();
-                    let path = metadata_entry.path();
-                    let target = Path::new(std::str::from_utf8(bytes).unwrap());
-                    let mode = metadata.mode().unwrap_or(0o0755) & 0o7777;
-                    let (seconds, _) = metadata.mtime().unwrap_or((0, 0));
-                    println!(
-                        "s {:o} {:>12} {:>16} {:?} -> {:?}",
-                        mode, seconds, "-", path, target
-                    );
-                }
-                item.into_inner()
+            // presense of metadata can be used to determine if this is the first apperiance of
+            // a directory by looking at the metadata: sibling hamt shard buckets do not have
+            // metadata.
```

Comment:
since this comment is still present ... how exactly are non-root buckets differentiated from directories?

Reply:
aah, it would be `ContinuedWalk::Directory(Entry::Bucket(_))`, I'll add a suggestion below